proof/access-control/RISCV64/ArchFinalise_AC.thy

```diff
@@ -241,7 +240,7 @@ global_interpretation Finalise_AC_1?: Finalise_AC_1
 proof goal_cases
   interpret Arch .
   case 1 show ?case
-    by (unfold_locales; (fact Finalise_AC_assms | wp finalise_cap_replaceable))
+    by (unfold_locales; wpsimp wp: Finalise_AC_assms)
```

Comment:
This is a strange formulation, it implies that `Finalise_AC_assms` is only `wp` rules. The previous formulation said: "if it isn't a fact then it's a wp rule".

Reply:
This formulation lets `wp` use the ones that are rules and will apply them as rule otherwise. I don't really mind, but it is true that the old version showed slightly more intention. Then again, it would also use `fact` with `Finalise_AC_assms` rules that are `wp` rules. So it didn't *really* make that distinction.

Reply:
So what this is doing is making use of rules declared `[wp]` in Arch in order to discharge locale assumptions without having to mark the relevant rules `[whatever_assms]`? I guess that makes sense; in theory if you put all the assms into the wp set, then `wpsimp` would solve all subgoals without further arguments?

Reply:
It definitely was not obvious at first glance though, and even now I seek confirmation. If we want to go down this route, we might want to say something in the arch-split docs.

Reply:
Agree that this isn't the ideal formulation. The issue is that some arch-specific `wp` rules have weaker preconditions than the locales, so `fact` isn't good enough. I'm still trying to think of a better solution.